au4/src/au3wrap/internal/domau3types.h

```diff
@@ -0,0 +1,15 @@
+/*
+* Audacity: A Digital Audio Editor
+*/
+#pragma once
+
+#include "libraries/lib-wave-track/WaveClip.h"
+
+namespace au::au3 {
+struct WaveClipID {
+    uint64_t id = -1;
+
+    WaveClipID(const WaveClip* au3clip)
+        : id(reinterpret_cast<uint64_t>(au3clip)) {}
```

Comment:
Since we are using pointers as id, it might make sense to use `using WaveClipID::type = uintptr_t` here for clarity, though it is effectively the same as uint64_t on most platforms.

Reply:
I thought about it, but I don't want to reveal the information that it is a pointer, i.e. that this pointer is a hack and just a current implementation detail